base_classes/NXroot.nxdl.xml

```diff
@@ -46,13 +46,35 @@
     <attribute name="file_update_time" type="NX_DATE_TIME">
         <doc>Date and time of last file change at close</doc>
     </attribute>
-    <attribute name="NeXus_version">
+    <attribute name="NeXus_version" deprecated="NAPI is frozen.">
         <doc>
             Version of NeXus API used in writing the file.
 
-            Only used when the NAPI has written the file.
             Note that this is different from the version of the
-            base class or application definition version number.
+            base class or application definition version number.            
+        </doc>
+    </attribute>
+    <attribute name="NeXus_repository">
+        <doc>
+            A repository containing the application definitions
+            used for creating this file.
+            If the NeXus_release attribute contains a commit distance and hash,
+            this should refer to this repository.
+        </doc>
+    </attribute>
+    <attribute name="NeXus_release">
+        <doc>
+            The version of NeXus definitions used in writing the file. This can either be a date-based
+            NeXus release (e.g., YYYY.MM), see https://github.com/nexusformat/definitions/releases or
+            a version tag that includes additional development information, such as a commit distance and
+            a Git hash. This is typically formatted as  `vYYYY.MM.post1.dev&lt;commit-distance&gt;.g&lt;git-hash&gt;`,
+            where `YYYY.MM` refers to the base version of the NeXus definitions. `post1.dev&lt;commit-distance&gt;` 
+            indicates that the definitions are based on a commit after the base version (post1), with
+            `;commit-distance&gt;` being the number of commits since that version. `g&lt;git-hash&gt;` is the 
```

Comment:
```suggestion
            `&lt;commit-distance&gt;` being the number of commits since that version. `g&lt;git-hash&gt;` is the 
```

Reply:
Also is that `.g<git-hash>` suffix widely used?

Reply:
In FAIRmat, we are using it when writing our NeXus files so that we can provide traceability back to the really exact commit in the repository. It's also a fairly standard wait of writing versions in git (see [here](https://git-scm.com/docs/git-describe) using the `--long` flag.

Reply:
The `git-describe` man page uses a `-g<git-hash>` suffix

Reply:
You are right, I think I missed your point here, sorry. Changed it to the ´ -g<git-hash>` notation (this is indeed the one we are also using).